### DISCUSSION TITLE
Question about auto save

### DISCUSSION THREAD

**Discussion by augustocdias** [rmagatti/auto-session#91] [Q&A] [answered]
I saw that in the docs:

```
Please note that if there are errors in your config, restoring the session might fail, if that happens, auto session will then disable auto saving for the current session. Manually saving a session can still be done by calling :SaveSession.
```

Will it go back to normal next time I open neovim without any errors? Will it start auto saving again or the session is corrupted and I have to recreate it?

**Comment by rmagatti** [comment, marked as answer]
@augustocdias yeah it definitely should. Auto Save is only disabled temporarily for that particular session so that it is less likely that one gets a corrupted session saved.

**Reply by augustocdias** [reply]
Then I guess I'm having an issue. It is only saving manually...

My configs:

```lua
require('auto-session').setup({
    auto_save_enabled = true,
    auto_restore_enabled = true,
    auto_session_enable_last_session = false,
    auto_session_allowed_dirs = { os.getenv('HOME') .. '/projects' },
})
require('session-lens').setup({})
vim.o.sessionoptions = 'buffers,curdir,folds,help,tabpages,winsize,resize,winpos,terminal'
```

**Reply by rmagatti** [reply]
If you are saving in your projects directory like you specified in allowed_dirs that should be working.
PS: you don't really need to say auto_save_enabled or auto_restore_enabled true, or last_session false for that matter, those are already the default values.

**Reply by rmagatti** [reply]
One thing to try is `:DeleteSession` then allow auto-session to create your session again. And see if that'll restore the auto-saving functionality.

**Reply by augustocdias** [reply]
It is not working... I've read the code and I understood that it auto saves when you close nvim right?

Here's a recording of it happening.

![tty](https://user-images.githubusercontent.com/1080789/142384131-34810657-ab98-4452-9571-d9f7c0b695a9.gif)

**Reply by rmagatti** [reply]
I think it's best if you open an issue about this so we can dig further. Maybe set the debug level to 'debug' and show me the logs you get there.
